net/v2/ipfamily.go

```diff
@@ -55,29 +95,56 @@ func IsDualStackIPs(ips []net.IP) bool {
 	return (v4Found && v6Found)
 }
 
-// IsDualStackIPStrings returns true if:
-// - all elements of ips can be parsed as IPs
-// - at least one IP from each family (v4 and v6) is present
-func IsDualStackIPStrings(ips []string) bool {
-	parsedIPs := make([]net.IP, 0, len(ips))
-	for _, ip := range ips {
-		parsedIP := ParseIPSloppy(ip)
-		if parsedIP == nil {
-			return false
+// IsDualStackPair returns true if vals contains exactly 1 IPv4 address and 1 IPv6 address
+// (in either order).
+func IsDualStackPair[T ipOrString](vals []T) bool {
```

Comment:
This is new. I kept noticing places in k/k that wanted this (half of https://github.com/danwinship/kubernetes/commit/4a88052a89cebe7b209bbdaa5a27c4d100f8cfb9)